lib/rspec/core/configuration.rb

```diff
@@ -778,24 +780,58 @@ def full_backtrace=(true_or_false)
         @backtrace_formatter.full_backtrace = true_or_false
       end
 
-      # Returns the configuration option for color, but should not
-      # be used to check if color is supported.
+      # Enables color output if the output is a TTY.  As of RSpec 3.6, this is
+      # the default behavior and this option is retained only for backwards
+      # compatibility.
       #
+      # @deprecated No longer recommended because of complex behavior. Instead,
+      #   rely on the fact that TTYs will display color by default, or set
+      #   {#color_mode} to :forced to display color on a non-TTY output.
+      # @see color_mode
       # @see color_enabled?
       # @return [Boolean]
       def color
         value_for(:color) { @color }
       end
 
+      # Sets the color mode to one of:
+      #
+      # - :automatic - the output will be in color if the output is a TTY (the
+      #   default)
+      # - :forced - the output will be in color, whether or not the output is a
+      #   TTY
+      # - :disabled - the output will not be in color
+      #
+      # @see color_enabled?
+      # @return [Boolean]
+      def color_mode
+        value_for(:color_mode) { @color_mode }
+      end
+
       # Check if color is enabled for a particular output.
       # @param output [IO] an output stream to use, defaults to the current
       #        `output_stream`
       # @return [Boolean]
       def color_enabled?(output=output_stream)
-        output_to_tty?(output) && color
+        if color_mode == :forced
+          true
+        elsif color_mode == :disabled
+          false
+        elsif color
+          tty? || output_to_tty?(output)
+        else # color_mode == :automatic
+          output_to_tty?(output)
+        end
       end
 
+      # Set the color mode.
+      attr_writer :color_mode
+
       # Toggle output color.
+      #
+      # @deprecated No longer recommended because of complex behavior. Instead,
+      #   rely on the fact that TTYs will display color by default, or set
+      #   {:color_mode} to :forced to display color on a non-TTY output.
```

Comment:
In general, I don't see a benefit to hard-deprecating something like this (e.g. with a warning).  A soft deprecation (via docs and a changelog entry) is sufficient for now, IMO, and then we can hard-deprecate it in 3.99.

In the 2.x releases we definitely got complaints about too many deprecation warnings.  In 3.x releases (where we've only been soft-deprecating things so far) I've not heard a single user complain that they wish we were issuing deprecation warnings.